src/hotspot/share/opto/regmask.hpp

```diff
@@ -63,29 +81,306 @@ class RegMask {
+  // words) that has been extended with two additional words to represent more
+  // stack locations:
+  //
+  //                         _lwm=1   RM_SIZE_IN_WORDS=3 _hwm=3      _rm_size_in_words=5
+  //                            |                  |      |                 |
+  //            r0 r1 r2 r3 r4 r5 r6 r7 r8 r9 s0 s1   s2 s3 s4 s5 s6 s7 s8 s9 s10 s11 ...
+  // Content:  [0  0  0  0 |0  1  1  0 |0  0  1  0 ] [1  1  0  1 |0  0  0  0] as  as  as
+  //   Index: [0]         [1]         [2]           [0]         [1]
+  //
+  //          \____________________________________/ \______________________/
+  //                                 |                           |
+  //                             _rm_word                     _rm_word_ext
+  //          \_____________________________________________________________/
+  //                                          |
+  //                                  _rm_size_in_words=5
+  //
+  // In this example, registers {r5, r6} and stack locations {s0, s2, s3, s5}
+  // are included in the register mask. Depending on the value of
+  // _infinite_stack (denoted with as), {s10, s11, ...} are all included (as=1)
+  // or excluded (as=0). Note that all registers/stack locations under _lwm
+  // and over _hwm are excluded. The exception is {s10, s11, ...}, where the
+  // value is decided solely by _infinite_stack, regardless of the value of
+  // _hwm.
+  //
+  // The only operation that may update the _offset attribute is
+  // RegMask::rollover(). This operation requires the register mask to be
+  // clean/empty (all zeroes), except for _infinite_stack, which must be true,
+  // and has the effect of increasing _offset by _rm_size_in_words and setting
+  // all bits (now necessarily representing stack locations) to 1. Here is how
+  // the above register mask looks like after clearing, setting _infinite_stack
+  // to true, and successfully rolling over:
+  //
+  //              _lwm=0                             RM_SIZE_IN_WORDS=3              _hwm=4  _rm_size_in_words=5
+  //                 |                                        |                        |      |
+  //            s10 s11 s12 s13 s14 s15 s16 s17 s18 s19 s20 s21  s22 s23 s24 s25 s26 s27 s28 s29 s30 s31 ...
+  // Content:  [1   1   1   1  |1   1   1   1  |1   1   1   1 ] [1   1   1   1  |1   1   1   1]  1   1   1
+  //   Index: [0]             [1]             [2]              [0]             [1]
+  //
+  //          \_______________________________________________/ \_____________________________/
+  //                                    |                                     |
+  //                                _rm_word                             _rm_word_ext
+  //          \_______________________________________________________________________________/
+  //                                                  |
+  //                                  _rm_size_in_words=_offset=5
```

Comment:
Can you please add some concise comment why we need `rollover`? Does that happen during register allocation, and if we have rollover then we start spilling instead of keeping values in registers?